### PR TITLE
Add some missing documentation to Assembly, some BCs, constraints, dampers, DGKernels, diracKernels, FEProblemSolve, functions and FVKernels

### PR DESCRIPTION
**Pull request by GiudGiud**
Big drive to Canyonlands, much documentation
Thanks Ryan for driving

Would love some input from @lindsayad on the assembly & constraints doc, @YaqiWang for DGKernels, @r for FVKernels

This will need another commit to address link failures

### CODE REVIEW

**Comment by loganharbour** [comment]
I assume you weren't the one that drove on your trip this weekend given this PR? 😆

File: framework/doc/content/source/executioners/FEProblemSolve.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
-!! MOOSE Documentation Stub: Remove this line when content is added.
-
 # FEProblemSolve.md
+
+!syntax description /Executioner/FEProblemSolve
+
+The `FEProblemSolve` class have two main roles: handle a variety of parameters for the libear and non linear solves
```
Comment (GiudGiud) [inline framework/doc/content/source/executioners/FEProblemSolve.md:5]:
```suggestion
The `FEProblemSolve` class has two main roles: handle a variety of parameters for the libear and non linear solves
```

File: framework/doc/content/source/base/Assembly.md [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+- the neighbor element
+
+- the local volumes
+
+- the Jacobian weights
+
+- the local quadrature rules and points
+
+- the local values of current variable, its spatial derivatives at first and second order,
+  its curl on the element
```
Comment (lindsayad) [inline framework/doc/content/source/base/Assembly.md:18]: Do you want these new lines between every list item? Does that even format correctly?

**Review by lindsayad — COMMENTED** [review]
`Assembly` content looks like a good starting point. Constraint documentation looks fine

File: framework/doc/content/source/base/Assembly.md (hunk 2 above)
Comment (GiudGiud) [reply framework/doc/content/source/base/Assembly.md:18]: formats fine https://mooseframework.inl.gov/docs/PRs/18866/site/source/base/Assembly.html
I ve found that more new lines work better than less with those lists tbh

Comment (lindsayad) [reply framework/doc/content/source/base/Assembly.md:18]: better how?

Comment (GiudGiud) [reply framework/doc/content/source/base/Assembly.md:18]: if you delete the new line before or one of the new lines after it wont work. It ll work fine within maybe

Comment (GiudGiud) [reply framework/doc/content/source/base/Assembly.md:18]: what would you add to Assembly?

Comment (lindsayad) [reply framework/doc/content/source/base/Assembly.md:18]: Assembly is many thousands of lines of code haha. But I would hope that our users don't often have to touch assembly methods, if ever. So I'm not sure whether we need a lot of user documentation for that object. Doxygen and code comments are more important IMO. In that vein, I think the intro material you have here is good

**Comment by GiudGiud** [comment]
pointed today two Discussions to two pages of documentation I wrote in the two previous PRs, these road trips are paying off lol

File: framework/doc/content/source/base/Assembly.md (hunk 2 above)
Comment (aeslaughter) [reply framework/doc/content/source/base/Assembly.md:18]: A one or zero new lines between list items should not render any differently. If you put two new lines you get a new list.

**Review by aeslaughter — DISMISSED** [review]

File: framework/doc/content/source/base/Assembly.md (hunk 2 above)
Comment (GiudGiud) [reply framework/doc/content/source/base/Assembly.md:18]: I m happy with a single new line. Unless you guys feel differently I was not planning on changing this

Comment (lindsayad) [reply framework/doc/content/source/base/Assembly.md:18]: I feel very strongly. Jk

**Comment by lindsayad** [comment]
@GiudGiud are you going to resolve your own conversation?

**Comment by GiudGiud** [comment]
This cant be merged as is there's a couple broken links. 
I will do it eventually, as you know I m working on a report right now ^^

**Comment by GiudGiud** [comment]
Ready for review now. 
Would be great if the syntax/children worked for abstract (in an input-file way) classes

**Comment by GiudGiud** [comment]
Someone with DGKernel experience could look at the DGDiffusion documentation?
There s a kernel for diffusion and a DGKernel for diffusion, not sure why is that. I wrote that down in the docs for it lol

**Comment by aeslaughter** [comment]
>  Would be great if the syntax/children worked for abstract (in an input-file way) classes

The available syntax is populated when a class is registered with `registerMooseObject`, perhaps a new macro could be created to "register" objects that can't be created, but it would take some work.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**